### PR TITLE
Unstick chatcommand! Will attempt to place you in the nearest standable spot. just type !stuck.

### PR DESCRIPTION
**Pull request by PYROTEKNIK**

### CODE REVIEW

File: lua/autorun/player_extension.lua [hunk 1]
```diff
@@ -155,4 +155,111 @@ function PLAYER:UsingWeapon(cls)
...
+    end
+    tr.mins = mins
+    tr.maxs = maxs
+    local trace = util.TraceHull(tr)
+    if(trace.StartSolid)then return true end
+    return false
+end
+
+if (SERVER) then
+    concommand.Add("stuck", function(ply, cmd, args)
```
Comment (swampservers) [inline lua/autorun/player_extension.lua]: What if this gets spammed?

Comment (PYROTEKNIK) [reply lua/autorun/player_extension.lua]: The command is throttled, but I could probably optimize this further

Comment (PYROTEKNIK) [reply lua/autorun/player_extension.lua]: wait no that's not supposed to still be there

Comment (swampservers) [reply lua/autorun/player_extension.lua]: i mostly mean make sure there is a rate limit on it (10 sec cooldown)

File: lua/autorun/player_extension.lua [hunk 2]
```diff
@@ -155,4 +155,103 @@ function PLAYER:UsingWeapon(cls)
     local c = self:GetActiveWeapon()
 
     return IsValid(c) and c:GetClass() == cls
+end
```
Comment (swampservers) [inline lua/autorun/player_extension.lua]: Can you just put all this in chatcommands/stuck.lua please?

Comment (PYROTEKNIK) [reply lua/autorun/player_extension.lua]: yeah one sec

File: lua/autorun/player_extension.lua (hunk 1 above)
Comment (PYROTEKNIK) [reply lua/autorun/player_extension.lua]: well i use whatever throttling fedorachat has. it also refuses to do any of the unstick functions if it doesn't think you're stuck

**Pull request merged**